src/target/stm32f1.c

```diff
@@ -679,6 +682,9 @@ static bool stm32f1_flash_write(target_flash_s *flash, target_addr_t dest, const
 	const size_t offset = stm32f1_bank1_length(dest, len);
 	DEBUG_TARGET("%s: at %08" PRIx32 " for %zu bytes\n", __func__, dest, len);
 
+	/* Allow wider writes on Gigadevices (is there a better way to hook this?) */
```

Comment:
This looks like a prime case to introduce a target-specific TOPT (`STM32F1_TOPT_GIGADEVICE`?) which the appropriate GD32* probe routines set

Reply:
Added a bit per suggestion, unable to test. Now that I think about it, I want Arterytek AT32F403A (and by extension AT32F415) also benefit from 32-bit writes, because they are definitely capable of this per their reference manuals. Can I rename this bit flag to something like `STM32F1_TOPT_32BIT_WRITES`? This sounds more like JLink FW capability bit flags now.